### ISSUE TITLE
[BUG] "consent required" when trying to log in

### ISSUE THREAD

**Issue by tdbe** [austinhuang0131/barinsta#1272]
I created a new insta account, and verified it with email and phone number, but didn't add any posts or any other insta activity (mentioning this in case it matters somehow). The account works when I log in from my browser.

In Barinsta, I go to Add Account, I go log in, after I enter my username and password and click login, it goes back to the Barinsta screen and there is a toast message popping up saying "consent required". And the login fails. ..But actually Barinsta thinks I am logged in. However, I do not see my username anywhere, and the Feed, Profile, and Discover pages are all blank and can't be refreshed. I can "log out" and try again, but the same thing happens. Tried force-closing the app etc.

Could it be that some extra step is required by instagram and Barinsta isn't showing it to me? Like an sms code? (it doesn't send me any emails or smses when I try to log in via barinsta though)

**Comment by austinhuang0131** [comment]
Are you European?

**Comment by tdbe** [comment]
> Are you European?

Yes. Maybe a GDPR popup then. I suppose all you need to test this is a VPN (though you maybe also need an european sim card)

**Comment by austinhuang0131** [comment]
Okay, maybe try going to browser and agree to whatever it needs you to agree on?

If you know what you're doing, can you also try capturing the request using dev panel when you click the agree button?

**Comment by tdbe** [comment]
Ok so when I log in via barinsta, as soon as I click log in after writig my password, the login webpage returns to the barinsta app so there's no opportunity to accept anything. However, I also logged into insta in my browser, and even though the cookies / sessions are not connected (between the browser I used, and barinsta's webview browser), now coming back to barinsta it works.

**Comment by austinhuang0131** [comment]
It's an account state so yeah it should work once you have accepted it anywhere... I'm asking if you accepted some dialog after logging into the browser one

**Comment by tdbe** [comment]
> It's an account state so yeah it should work once you have accepted it anywhere... I'm asking if you accepted some dialog after logging into the browser one

Ah. When I first logged in, before trying Barinsta at all, it was on my PC, in a browser, and I must have agreed on everything, because I remember being able to look through instagram as you normally do while logged in. And then I went to barinsta and had the thing I explained, and then I went again in my browser (but on my phone this time) and agreed again, and then it worked. Not sure what happened. But it's probably not Barinsta's fault, it's probably instagram being annoying with their TOS and selling your soul to the zuckerbergs etc.
